src/app/manageinvoices/manageinvoices.component.ts

```diff
@@ -7,63 +13,102 @@ import { Component, OnInit } from '@angular/core';
+    }
+    else if (MyRole?.includes(UserRole.CUSTOMER))
+    {
+      this.invoiceService.GetUserInvoices().subscribe(invoices => {this.invoices = invoices; console.log(this.invoices);});
+      this.UserMan = true;
+    }    
+    else
+    {
+      this.LogonError = true;
+    }
+  }
+ 
+  dateFormat(date : Date) : string
+  {
+    return formatDate(date, 'MM/dd/yyyy', 'en');
   }
 
-  public open(id: number): void {
-    let invoice = new Invoice();
-    this.invoices.forEach(function (inv) {
-      if (inv.id == id) {
-        invoice = new Invoice(inv.id, inv.CustomerName, inv.SupplierID.toString(), inv.Date, inv.DueDate, inv.GasAmount, inv.ElectricityAmount, inv.Price, inv.Tax, "", inv.StatusID, inv.Status);
-      }
-    });
-    this.invoicedetails = invoice;
-    return;
+  GetCustomerType(customer : CustomerType) : string
+  {
+    return CustomerType[Number(customer)];
+  }
+
+  GetInvoiceType(customer : INVOICE_TYPE) : string
+  {
+    return INVOICE_TYPE[Number(customer)];
+  }
+  
+  GetInvoiceStatus(customer : INVOICE_STATUS) : string
+  {
+    return INVOICE_STATUS[Number(customer)];
+  }
+
+  CheckInvoiceStatus(customer : INVOICE_STATUS) : boolean
+  {
+    if(customer == INVOICE_STATUS.PAID)
+    {
+      return false;
+    }
+    return true;
   }
-}
 
-export class Invoice {
-  public id: number;
-  public customerName: string;
-  public supplierName: string;
-  public date: string;
-  public dueDate: string;
-  public gasAmount: number;
-  public electricityAmount: number;
-  public price: number;
-  public tax: number;
-  public contractType: string;
-  public statusID: number;
-  public status: string;
+  OnClickDetails(i : number)
+  {
+    this.DetailsID = i;
+    this.DetailsClicked = true;
+    this.DownloadPDF()
+  }
+
+  CheckUserMan(i : number | undefined) : boolean
+  {
+    console.log(i);
+    if(i != this.auth.getUserId())
+    {
+      return false;
+    }
+    return true;
+  }
 
-  constructor (iId: number = 0, iCName: string = "", iSName = "", iDate = "", iDueDate = "", iGasAmount = 0, iElectricityAmount = 0, iPrice = 0, iTax = 0, iContractType = "", iStatusID = 0, iStatus = "") {
-    this.id = iId;
-    this.customerName = iCName;
-    this.supplierName = iSName;
-    this.date = iDate;
-    this.dueDate = iDueDate;
-    this.gasAmount = iGasAmount;
-    this.electricityAmount = iElectricityAmount;
-    this.price = iPrice;
-    this.tax = iTax;
-    this.contractType = iContractType;
-    this.statusID = iStatusID;
-    this.status = iStatus;
+  DownloadPDF()
+  {
+    this.invoiceService.downloadInvoiceFile(this.invoices[this.DetailsID].id).subscribe(response => {
+      let blob: Blob = response.body as Blob;
+      window.open(window.URL.createObjectURL(blob));
+    });
+  }
+
+  UpdatePayment(i : number)
+  {
+    this.invoiceService.UpdateInvoice(this.invoices[i]).subscribe(response => {
+      console.log("Invoice Updated");
+      window.location.reload();
+    });
   }
```

Comment:
What I recommend:
in invoice service you create a public variable which u access inside your component via this.invoiceService.invoices.
You use functions to update the invoices. After updating the payment, you run the function to update the invoices inside the service again. Angular will notice that the contents changed and will update ngIf's ngFors etc. No need to reload at that point anymore.